lib/openshift_client.rb

```diff
@@ -20,11 +20,28 @@ class Client < Kubeclient::Common::Client
       [OpenshiftClient.const_set(et, Class.new(RecursiveOpenStruct)), et]
     end
 
-    def initialize(uri, version = 'v1', path = '/oapi')
+    def initialize(uri,
+                   version = 'v1',
+                   path = '/oapi',
+                   ssl_options: {
+                     client_cert: nil,
+                     client_key: nil,
+                     ca_file: nil,
+                     verify_ssl: OpenSSL::SSL::VERIFY_PEER
+                   },
+                   auth_options: {})
+      fail ArgumentError, 'Missing uri' if uri.nil?
+
+      validate_auth_options(auth_options)
+
       handle_uri(uri, path)
       @api_version = version
       @headers = {}
-      ssl_options
+      @ssl_options = ssl_options
+
+      @basic_auth_user = auth_options[:user]
```

Comment:
On Tue, Aug 25, 2015 at 2:50 PM, abonas <notifications@github.com> wrote:

> In lib/openshift_client.rb
> <https://github.com/abonas/openshift_client/pull/10#discussion_r37856078>:
>
> >        handle_uri(uri, path)
> >        @api_version = version
> >        @headers = {}
> > -      ssl_options
> > +      @ssl_options = ssl_options
> > +
> > +      @basic_auth_user = auth_options[:user]
>
> Pretty sure those changes where made in the same pr.
>
> let me know if you need anything from me to move this patch.
>
> please look here
> https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51
> as you can see, there's a conditional there that checks if basic auth
> details should be assigned or not.
> in your current patch here, you are always assigning basic auth details,
> while the bearer token is assigned conditionally - hence the logic is
> different.
>

That change in logic was done as part of dyson bearer patch
(not saying it's related to the bearer patch but it was 'snuck in' together
)

so it is an additional change.

—
> Reply to this email directly or view it on GitHub
> <https://github.com/abonas/openshift_client/pull/10/files#r37856078>.
>